### PR TITLE
TECH-1861 - Migrating cla-assistant to Kubernetes

### PR DESCRIPTION
**Pull request by OleksandrUA**
TECH-1861 - Migrating cla-assistant to Kubernetes

### CODE REVIEW

**Review by sanbotto — APPROVED** [review]

File: .github/workflows/build-and-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        LATEST_TAG: latest
+        ENVIRONMENT_TAG: prod
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+      run: |  
+        # Build Docker containers and push them to ECR cla-assistant-prod
+        docker pull $ECR_REGISTRY/cla-assistant-prod:$LATEST_TAG || true
+        docker build -t cla-assistant-prod \
+                     -t $ECR_REGISTRY/cla-assistant-prod:$SHA_TAG \
+                     -t $ECR_REGISTRY/cla-assistant-prod:$LATEST_TAG \
+                     -t $ECR_REGISTRY/cla-assistant-prod:$ENVIRONMENT_TAG \
```
Comment (eskp) [inline .github/workflows/build-and-deploy.yaml:51]: Do you want to remove "prod" from the service name? You have ENVIRONMET_TAG for that

File: .github/workflows/build-and-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      - main
+
+name: Deploy to AWS Production (K8s)
+
+jobs:
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    environment: production
+    env:
```
Comment (eskp) [inline .github/workflows/build-and-deploy.yaml:13]: I suggest putting SERVICE_NAME and maybe ENVIRONMENT_TAG up here as well. So it's not needed to edit this file all that much between services and environments, therefore less chances for typos

File: deploy/prod/cla-assistant.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  enabled: true
+  name: cla-assistent
+  port: 8080
+  type: ClusterIP
+  containerPort: 8080
+  tls:
+    enabled: true
+    issuerName: cloudflare
+image:
+  repository: ${ECR_REGISTRY}/cla-assistent-prod
```
Comment (eskp) [inline deploy/prod/cla-assistant.yaml:12]: prod should be a tag

File: deploy/prod/cla-assistant.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  create: false
+ingress:
+  enabled: true
+  host: cla.daifoundation.org
+  annotations:
+    external-dns.alpha.kubernetes.io/cloudflare-proxied: "true"
+podAnnotations:
+  reloader.stakater.com/auto: "true"
+resources:
+  limits:
```
Comment (eskp) [inline deploy/prod/cla-assistant.yaml:25]: Huge limit for both cpu and mem?

**Review by eskp — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-deploy.yaml (hunk 1 above)
Comment (OleksandrUA) [reply .github/workflows/build-and-deploy.yaml:51]: @eskp that is also the ECR registry name, and unfortunately ECR names are immutable. I'd prefer to not destroy all previous images.

File: deploy/prod/cla-assistant.yaml (hunk 4 above)
Comment (OleksandrUA) [reply deploy/prod/cla-assistant.yaml:25]: Yes, on ECS these were even higher, I adjusted according to real monitoring data

File: deploy/prod/cla-assistant.yaml (hunk 3 above)
Comment (OleksandrUA) [reply deploy/prod/cla-assistant.yaml:12]: @eskp sorry, didn't get it. `prod` is set as tag now, have a look on line `14`

File: .github/workflows/build-and-deploy.yaml (hunk 2 above)
Comment (OleksandrUA) [reply .github/workflows/build-and-deploy.yaml:14]: @eskp done

File: .github/workflows/build-and-deploy.yaml (hunk 1 above)
Comment (eskp) [reply .github/workflows/build-and-deploy.yaml]: There are only 14 tags of ~100MB in size. You could pull them all with `-a` flag then push all up to the new repo.

If you can't be bothered I think it's also quite fine to get rid of all of them and start with a fresh repo.

File: deploy/prod/cla-assistant.yaml (hunk 3 above)
Comment (eskp) [reply deploy/prod/cla-assistant.yaml:12]: This is related to changing the ECR name and getting rid of "prod" in the name

File: .github/workflows/build-and-deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+      run: |
+        sed -i 's/${ECR_REGISTRY}/'$(echo $ECR_REGISTRY)/ deploy/prod/cla-assistant.yaml
+
+    - name: Deploying CLA Assistant Logger to Kubernetes with Helm
+      uses: bitovi/github-actions-deploy-eks-helm@v1.2.4
+      with:
+        values: image.repository=${{ steps.login-ecr.outputs.registry }}/${{ env.AWS_ECR_NAME }},image.tag=${{ steps.vars.outputs.sha_short }}
+        cluster-name: ${{ env.CLUSTER_NAME }}
+        config-files: deploy/prod/cla-assistant.yaml
```
Comment (eskp) [inline .github/workflows/build-and-deploy.yaml:72]: Cloud use `SERVICE_NAME` here (and on line 74) also if you like

Comment (OleksandrUA) [reply .github/workflows/build-and-deploy.yaml:72]: @eskp now path looks clean and I know where to find the file, adding var to file path doesn't sound like improvement to me.
As for the 74 it contains namespace name not the service name so I would prefer to keep things clear and explicit.

File: .github/workflows/build-and-deploy.yaml (hunk 1 above)
Comment (OleksandrUA) [reply .github/workflows/build-and-deploy.yaml]: @eskp migrated all existing images from -prod to new one and changed CICD pipelines accordingly

**Comment by eskp** [comment]
Nice one, thanks Sasha

**Pull request merged**